File: numass-control/src/main/java/inr/numass/control/DeviceViewFactory.java
package inr.numass.control;

import hep.dataforge.control.devices.Device;
import hep.dataforge.control.devices.DeviceFactory;

public interface DeviceViewFactory extends DeviceFactory {
    /**
     * Create but do not connect view connection for the device
     * @return
     */
    DeviceViewConnection buildView(Device device);
}

File: numass-control/src/main/kotlin/inr/numass/control/DeviceViewFactory.kt
package inr.numass.control

import hep.dataforge.control.devices.Device
import hep.dataforge.control.devices.DeviceFactory

interface DeviceViewFactory<D : Device> : DeviceFactory<D> {
    /**
     * Create but do not connect view connection for the device
     * @return
     */
    fun buildView(device: Device): DeviceViewConnection<D>
}
